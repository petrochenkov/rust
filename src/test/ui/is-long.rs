// Copyright 2017 The Rust Project Developers. See the COPYRIGHT
// file at the top-level directory of this distribution and at
// http://rust-lang.org/COPYRIGHT.
//
// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
// option. This file may not be copied, modified, or distributed
// except according to those terms.

#![warn(unused)]

fn main() {
    let my_opt = Some(15);
    if my_opt is Some(x) &&
       x > 10 {
        println!("{:?}", x);
    }

    let mut iter = my_opt.into_iter();
    while iter.next() is Some(mut y) &&
          y > 10 {
        println!("{:?}", y);
    }

    my_opt is Some(z);

    my_opt is Some(unused); //~ WARN variable `unused` is assigned to, but never used
                            //~^ WARN value assigned to `unused` is never read

    let d = '6';
    let is_digit = d is '0' ... '9' && d > '5';
    if is_digit {
        println!("{:?}", d);
    }

    // Variables bound in `is` are in scope until the end of the block in this configuration
    let _x = x; //~ ERROR use of possibly uninitialized variable: `x`
    let _y = y; //~ ERROR use of possibly uninitialized variable: `y`
    let _z = z; //~ ERROR use of possibly uninitialized variable: `z`
}
