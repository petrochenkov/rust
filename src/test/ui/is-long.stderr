warning: unreachable pattern
  --> $DIR/is-long.rs:31:5
   |
31 | /     if Ok(10) is Ok(u) | Err(u) { //~ WARN unreachable pattern
32 | |         println!("{:?}", u);
33 | |     }
   | |_____^
   |
note: lint level defined here
  --> $DIR/is-long.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unreachable_patterns)] implied by #[warn(unused)]

warning: unreachable pattern
  --> $DIR/is-long.rs:35:5
   |
35 |       if 0u8 is w { //~ WARN unreachable pattern
   |       ^         - matches any value
   |  _____|
   | |
36 | |         || w;
37 | |     }
   | |_____^ unreachable pattern

warning: variable `unused` is assigned to, but never used
  --> $DIR/is-long.rs:28:20
   |
28 |     my_opt is Some(unused); //~ WARN variable `unused` is assigned to, but never used
   |                    ^^^^^^
   |
note: lint level defined here
  --> $DIR/is-long.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_variables)] implied by #[warn(unused)]
   = note: consider using `_unused` instead

warning: value assigned to `unused` is never read
  --> $DIR/is-long.rs:28:5
   |
28 |     my_opt is Some(unused); //~ WARN variable `unused` is assigned to, but never used
   |     ^^^^^^^^^^^^^^^^^^^^^^
   |
note: lint level defined here
  --> $DIR/is-long.rs:11:9
   |
11 | #![warn(unused)]
   |         ^^^^^^
   = note: #[warn(unused_assignments)] implied by #[warn(unused)]

error[E0381]: use of possibly uninitialized variable: `x`
  --> $DIR/is-long.rs:46:9
   |
46 |     let _x = x; //~ ERROR use of possibly uninitialized variable: `x`
   |         ^^ use of possibly uninitialized `x`

error[E0381]: use of possibly uninitialized variable: `y`
  --> $DIR/is-long.rs:47:9
   |
47 |     let _y = y; //~ ERROR use of possibly uninitialized variable: `y`
   |         ^^ use of possibly uninitialized `y`

error[E0381]: use of possibly uninitialized variable: `z`
  --> $DIR/is-long.rs:48:9
   |
48 |     let _z = z; //~ ERROR use of possibly uninitialized variable: `z`
   |         ^^ use of possibly uninitialized `z`

error[E0381]: use of possibly uninitialized variable: `u`
  --> $DIR/is-long.rs:49:9
   |
49 |     let _u = u; //~ ERROR use of possibly uninitialized variable: `u`
   |         ^^ use of possibly uninitialized `u`

error: aborting due to 4 previous errors

